userspace/libscap/examples/01-open/test.c

```diff
@@ -336,6 +350,120 @@ void print_parameter(int16_t num_param)
+
+	/* storing strings in the lexicographical order */
+	for(int i = 0; i < dim; ++i)
+	{
+		for(int j = i + 1; j < dim; ++j)
+		{
+			/* swapping strings if they are not in the lexicographical order */
+			if(strcmp(string_vector[i], string_vector[j]) > 0)
+			{
+				strcpy(temp, string_vector[i]);
+				strcpy(string_vector[i], string_vector[j]);
+				strcpy(string_vector[j], temp);
+			}
+		}
+	}
+
+	printf("\nSyscalls in the lexicographical order: \n");
+	for(int i = 0; i < dim; i++)
+	{
+		printf("[%d] %s\n", i, string_vector[i]);
+	}
+	printf("Interesting syscalls: %d\n", dim);
+}
+
+/* This are the real interesting syscalls that we want to support in the new probe.
+ * - all syscalls associated with events of type `UF_NEVER_DROP`.
+ * - all syscalls that are not managed through `GENERIC_EVENTS` and don't
+ *   have the `EF_DROP_SIMPLE_CONS` flag.
+ *
+ * Please note: if some syscalls miss probably you have an old kernel
+ * that don't define them. Try to use a newer one.
+ */
+
+void print_modern_probe_syscalls()
+{
+	char str[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_SIZE];
+	int interesting_syscall = 0;
+	enum ppm_syscall_code ppm_syscall_code = 0;
+
+	/* For every syscall of the system. */
+	for(int syscall_id = 0; syscall_id < SYSCALL_TABLE_SIZE; syscall_id++)
+	{
+		/* TAKE ALWAYS: If the syscall has `UF_NEVER_DROP` flag we cannot use simple consumer. */
+		if(g_syscall_table[syscall_id].flags & UF_NEVER_DROP)
+		{
+			ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+			if(!g_syscall_info_table[ppm_syscall_code].name)
+			{
+				printf("*ERROR: a `UF_NEVER_DROP` syscall has no the corresponding name in the g_syscall_info_table.\n");
+				exit(EXIT_FAILURE);
+			}
+			strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_syscall_code].name);
+			continue;
+		}
+
+		/* TAKE NEVER: If we use generic events, we can drop the syscall with the simple consumer logic. */
+		if(g_syscall_table[syscall_id].enter_event_type == PPME_GENERIC_E)
+		{
+			continue;
+		}
+
+		ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+		if(g_syscall_info_table[ppm_syscall_code].flags & EF_DROP_SIMPLE_CONS)
+		{
+			continue;
+		}
+
+		/* This is an error since it means that a syscall we want to trace is not tracked in our `g_syscall_info_table`.
+		 * We have `EC_UNKNOWN` when we don't have an entry in the `g_syscall_info_table`.
+		 */
+		if(g_syscall_info_table[ppm_syscall_code].category == EC_UNKNOWN)
+		{
+			printf("*ERROR: the syscall with ppm code '%d' has an event associated but it is unknown in our g_syscall_info_table.\n", ppm_syscall_code);
+			exit(EXIT_FAILURE);
+		}
+
+		strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_syscall_code].name);
+	}
+
+	print_ordered_syscalls(str, interesting_syscall);
+}
+
+/* Print all supported syscall. Please if you only want
+ * to see syscalls in simple consumer mode, enable the
+ * `--simple_consumer` option.
+ */
+void print_kernel_simple_consumer_syscalls()
```

Comment:
I'd rename this one to
```suggestion
void print_supported_syscalls()
```

Reply:
changed